### ISSUE TITLE
Corruption with high number of IP's in any_of

### ISSUE THREAD

**Issue by szibis** [grobian/carbon-c-relay#288]
```
[2017-06-28 14:31:21] starting carbon-c-relay v3.1 (5dbe62), pid=4707
*** Error in `/usr/bin/relay': double free or corruption (fasttop): 0x0000000000a3d0b0 ***
configuration:
    relay hostname = ip-10-1-9-192
    listen port = 2013
    workers = 32
    send batch size = 40000
    server queue size = 30000000
    server max stalls = 3
    listen backlog = 64
    socket bufsize = 425984
    server connection IO timeout = 3000ms
    configuration = /etc/carbon-c-relay/relay.conf
```

With any_of and more than 22 IP's inside I get this error.

Currently used version: carbon-c-relay v3.1 (5dbe62)

**Comment by grobian** [comment]
Does the config from the test in 1f7a02b crash for you?

**Comment by szibis** [comment]
No

**Comment by grobian** [comment]
Ok, can you share a bit more of your config then?

**Comment by grobian** [comment]
Could you perhaps get a backtrace?  Also testing the code including d993fa6 would perhaps help.

**Comment by grobian** [comment]
I think I fixed this in 50987b7
